src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -38,61 +38,64 @@ import { LoadingListener } from '../../utils/loading/loading-listener';
     trigger('validated', [
       transition(':enter', [
         animate(
-          '300ms',
+          '600ms',
           keyframes([
-            style({ transform: 'scale(0,0)' }),
-            style({ opacity: 1 }),
-            style({ transform: 'scale(1.2,1.2)' }),
-            style({ transform: 'scale(.9,.9)' }),
-            style({ transform: 'scale(1,1)' }),
+            style({ transform: 'scale(0,0)', offset: 0 }),
+            style({ opacity: 1, offset: 0.2 }),
+            style({ transform: 'scale(1.2,1.2)', offset: 0.4 }),
+            style({ transform: 'scale(.9,.9)', offset: 0.6 }),
+            style({ transform: 'scale(1,1)', offset: 1 }),
           ])
         ),
       ]),
       transition(':leave', [style({ opacity: 1 }), animate('100ms ease-out', style({ opacity: 0 }))]),
     ]),
   ],
+  host: { '[attr.disabled]': "disabled? '' : null" },
 })
 export class ClrLoadingButton implements LoadingListener {
   public buttonState = ClrLoadingState;
   public state: ClrLoadingState = ClrLoadingState.DEFAULT;
 
+  @ContentChild(ClrLoading) public loadingDirective: ClrLoading;
```

Comment:
`@ContentChild` is not what you want to do here. The loading directive and this component are on the same host, so you want to inject it directly in the constructor.

But I just realized you are not using this `loadingDirective` anywhere, did you forgot to remove it?

Reply:
When I directly invoke the loadingStateChange method in the template, I see the change emit twice. Is this the intended behavior or am I doing something wrong?